eZ/Bundle/EzPublishCoreBundle/Command/CopySubtreeCommand.php

```diff
@@ -0,0 +1,194 @@
+                'eZ Platform username (with Role containing at least Content policies: create, read)',
+                'admin'
+            )
+            ->setDescription('Copy subtree from one location to another');
+    }
+
+    /**
+     * @param \Symfony\Component\Console\Input\InputInterface $input
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        parent::initialize($input, $output);
+        $this->permissionResolver->setCurrentUserReference(
+            $this->userService->loadUserByLogin($input->getOption('user'))
+        );
+    }
+
+    /**
+     * @param \Symfony\Component\Console\Input\InputInterface $input
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     *
+     * @return int|null|void
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $sourceLocationId = $input->getArgument('source-location-id');
+        $targetLocationId = $input->getArgument('target-location-id');
+
+        $sourceLocation = $this->locationService->loadLocation($sourceLocationId);
+        $targetLocation = $this->locationService->loadLocation($targetLocationId);
+
+        $sourceContentType = $this->contentTypeService->loadContentType(
+            $sourceLocation->getContentInfo()->contentTypeId
+        );
+        if (!$sourceContentType->isContainer) {
+            throw new InvalidArgumentException(
+                'source-location-id',
+                'Cannot copy subtree of non container location'
+            );
+        }
+
+        $targetContentType = $this->contentTypeService->loadContentType(
+            $targetLocation->getContentInfo()->contentTypeId
+        );
+
+        if (!$targetContentType->isContainer) {
+            throw new InvalidArgumentException(
+                'target-location-id',
+                'Cannot copy location to a parent that is not a container'
+            );
+        }
+        $questionHelper = $this->getHelper('question');
+        $question = new ConfirmationQuestion(
+            sprintf(
+                'Are you sure you want to copy `%s` subtree (size: %d) into `%s`? This make take a while for big number of nested children. yes/no?',
+                $sourceLocation->contentInfo->name,
+                $this->getDeepChildCount($sourceLocation),
+                $targetLocation->contentInfo->name
+            )
+        );
+
+        if (!$questionHelper->ask($input, $output, $question)) {
+            return;
```

Comment:
If we're returning something, then rather 0 in this case (based on Symfony doc for `execute`).